frontend/lib/ui/NumberInput/NumberInput.test.tsx

```diff
@@ -38,4 +38,28 @@ describe("UI Component: number input", () => {
     expect(input).toHaveValue("12.500");
     expect(input.selectionStart).toBe(4);
   });
+
+  test("should backspace the right character", async () => {
+    render(<NumberInput id="test" defaultValue={12345} />);
+    const input: HTMLInputElement = screen.getByTestId("test");
+    const user = userEvent.setup();
+
+    input.focus();
+    //value should be 12.345
+    await user.keyboard("{arrowleft}{arrowleft}{arrowleft}");
+    await user.keyboard("{backspace}");
+    expect(input).toHaveValue("1.345");
+  });
+
+  test("should delete the right character", async () => {
+    render(<NumberInput id="test" defaultValue={12345} />);
+    const input: HTMLInputElement = screen.getByTestId("test");
+    const user = userEvent.setup();
+
+    input.focus();
+    //value should be 12.345
+    await user.keyboard("{arrowleft}{arrowleft}{arrowleft}{arrowLeft}");
+    await user.keyboard("{delete}");
+    expect(input).toHaveValue("1.245");
```

Comment:
Shouldn't the 2 be deleted instead of the 3 when you move the caret 4 times to the left? Or we count the dot as a character I guess?

Reply:
Good suggestion @Lionqueen94, not sure if we could implement/build that. But since you can't really type the `.` yourself, we might try and ignore it altogether when moving the caret. Just like we do now when using backspace or delete. Would mean that the cursor would jump over it when moving the caret.

Reply:
Exactly, that makes more sense to me since it's just a styling dot, that you cannot add or remove yourself.

Reply:
I think this is very doable.

Reply:
I left the other behavior in as wel since you can still "reach" the dot by mouse or by deleting a char.
it looks really weird if you click on "pos x" and the cursor jumps.

Reply:
@lkleuver (looks like the caret only 'jumps' now when you are moving with the left arrow, and not with the right arrow. Shouldn't this be both ways?

Reply:
mmm, jumping the cursor breaks the current selection range. 
repro:
- input 1234
- move cursor to last position
- press and hold shift
- use left arrow to expand selection 
- observed: after you pass the `.`, a new selection range is set (containing the `1.`)
- expected: selection range contains `1.234` after passing the dot.

Reply:
Related, using shift+arrow:

1. enter "12.345"
2. use arrows to navigate to just after the first character: `1|2.345`
3. press shift + arrow right

This results in the separator symbol being selected instead of the "2".

Reply:
Another one related to selection:

1. enter `123.456.789`
2. put your cursor before the `3`: `12|3.456.789`
3. select up to the `5`, so `.45` are selected
4. hit the delete key

This results in `12.356.789`, so the `5` is still there, even though it was selected and should have been deleted.

Reply:
I've changed is so formatting is removed on focus and returned on blur, also added an onKeyDown handler to make it behave as input type="number"